### ISSUE TITLE
Error in solve.default(cov, ...) : system is computationally singular: reciprocal condition number

### ISSUE THREAD

**Issue by gabocks** [marlonecobos/mop#3]
Hi there,

Thank you very much for this package and the associated paper, it gave me a great way to measure extrapolation accuracy.

Because some of my rasters have very small values, I ran into an error while using the "mop" function on my dataset: Error in solve.default(cov, ...) : system is computationally singular: reciprocal condition number

I found the solution to solve it here: https://stackoverflow.com/questions/22134398/mahalonobis-distance-in-r-error-system-is-computationally-singular

In short, I needed to be able to set the "tol" argument of the "solve" of the stats::mahalanobis function to a very small number (tol=1e-20). I used an easy workaround by changing setting tol within mop_distance and creating 2 slightly changed functions. But it means I need to load the script with the new functions. I think it would be great if the tol argument could be directly set up within mop(). Or if there was a check within that would transform the necessary rasters.

Best,
Gaelle

**Comment by marlonecobos** [comment]
Hi, 

Thanks for the suggestion. 

It is a good idea to include this tolerance argument in the functions to avoid this kind of problem. However, there will be times in which the tolerance value won't make a difference.  Sometimes, no matter what you do, the cov matrix derived from the data won't be positive definite, which will generate a problem. If you want to avoid that error, using Euclidean distances will be your only option if you don't want to transform your data.  

I will include this change in the next version of the package (soon).

Best,

**Comment by gabocks** [comment]
Hey,

Thanks for the information, I am very new to multivariate distance calculation. Maybe if there could be an error warning (from within mop) "error the cov matrix is not positive you need to use Euclidian distance". It might help users with less skills that would otherwise be stuck?

Anyway, thank you for the quick answer and great package.

Cheers

**Comment by marlonecobos** [comment]
Done!
